### ISSUE TITLE
The applications 'undefined' was not modified

### ISSUE THREAD

**Issue by elGonzass** [OutSystems/vsts#11]
If i try to push two releases in VSTS without making changes in the selected OutSystems apps in between, the release fails with an "The applications 'undefined' was not modified. Maybe syncronizing your environments will do." error. Anything i can do about it?

Thanks,

Gonzalo

**Comment by falebarros** [comment]
Hi elGonzass,

Thanks for the feedback.
That is the way LifeTime Deployment API works. If you create a deployment plan but there are no changes to deploy, the deployment fails.

(I don't like the 'undefined' but seems to work as expected).
Do you see the need to:
-document that behavior?
-change to a different behavior? Why?

Thanks
Antonio

**Comment by elGonzass** [comment]
I'd like to schedule my CI / CD pipeline to run daily, but there might not be changes to all the apps in that particular day... so instead of failing i'd like the deployment to succeed, even if it means doing nothing since there's nothing to update

**Comment by sumo300** [comment]
Yep.  We just updated the extension and now get this error.  You should not be raising an error when there are no changes to be found that need to be moved to the next environment, regardless of how the Lifetime API works.  It should log that nothing was moved and that it succeeded.

Here's why.  If I were to take a normal .NET application and deploy it to an environment, I can still re-deploy the exact same version with no side effects.  I may want to do this for whatever reason.  This is not an error condition.  So, with that, I'd expect a deploy or re-deploy of the same version to be met with success rather than failure in the task itself.

Thoughts?

**Comment by falebarros** [comment]
Great feedback Andy and Michael.

I agree with you.
But, he have an issue here:
-Imagine you actually tag an application that has no changes. If you try to deploy it, you'll get that "applications 'undefined' was not modified" error.
We are thinking of ignoring that error, because is not a CI/CD error.
But, because we will not deploy app, the "tag version" is not propagated to target environment.
Are you OK with that (because forcing the deploying the new version is not available)?

By the way, are you deploying OutSystems application alone or the deployment includes other components non OutSystems?

Thanks
Antonio

**Comment by sumo300** [comment]
@falebarros At the moment, we're simply trying to deploy an OS app by itself.

**Comment by elGonzass** [comment]
In my case, we're deploying an OS app as the first step in a CI/CD pipeline, next steps will include automating testing and communication to other internal systems. @falebarros any thoughts on if/when the change would be ready? Thanks

**Comment by falebarros** [comment]
Hi,

Just published a new version. There was actually a bug that I fixed.
Unfortunately I didn't run all the tests. Sorry.

Cheers

**Comment by falebarros** [comment]
Hi Michael,

Have you tried to uninstall OutSystems Integration and install again?
I created a new VSTS area, with a new project; installed OutSystems Extension and created the deployment for an application without any change. It worked.

May you try that (uninstall Extension and install again), please?

Thanks
Antonio

**Comment by sumo300** [comment]
FYI @elGonzass the latest extension update has fixed this issue.  We tested it successfully on TFS 2018 Update 2.
